### PR TITLE
Loading component displayed when loading data from API

### LINKED ISSUE #654: Slow API response times can lead to inconsistent UX
### Current Behavior

When opening any page that hits the forge API to load data, the UI state can be inconsistent until the API responds and then the UI updates to match.

The following are the most obvious ones that need addressing (there may be others, but lets start here...)

1. Team Overview - this is the page most uses land on when they log in. It lists their projects - but until the API call to list the projects returns, the page says 'you have no projects'. That can be distressing to read when you expect lots of projects to be listed.
2. Team Projects page - as above
3. Team Devices page - same again, just devices no projects
4. Project Devices
5. Project Snapshots

### Expected Behavior

Some kind of loading placeholder so we don't show the wrong state whilst API calls are pending.

### Steps To Reproduce

Use Chrome Dev tools to throttle the network link.

### Environment

- FlowForge version:
- Node.js version:
- npm version:
- Platform/OS:
- Browser:

### PR DESCRIPTION
**Pull request by joepavitt**
Closes #654

### CODE REVIEW

File: frontend/src/pages/team/Projects.vue [hunk 1]
```diff
@@ -18,11 +18,12 @@
                 </ff-button>
             </div>
         </template>
-        <template v-else>
+        <template v-else-if="!loading">
```
Comment (knolleary) [inline frontend/src/pages/team/Projects.vue:21]: This condition needs moving to the top so it takes precedence over the others. As it stands, it is never being reached.

File: frontend/src/pages/team/Devices/index.vue [hunk 2]
```diff
@@ -37,6 +37,7 @@
                 </template>
             </div>
         </template>
+        <ff-loading v-else message="Loading Devices..." />
```
Comment (knolleary) [inline frontend/src/pages/team/Devices/index.vue:40]: This condition needs moving to the top so it takes precedence over the others. As it stands, it is never being reached.

**Review by knolleary — CHANGES_REQUESTED** [review]
A couple instances where the `v-if` logic isn't being reached to display the loading component.

File: frontend/src/pages/team/Devices/index.vue [hunk 3]
```diff
@@ -92,6 +93,7 @@ export default {
     name: 'TeamDevices',
     data () {
         return {
+            loading: false,
```
Comment (knolleary) [inline frontend/src/pages/team/Devices/index.vue:96]: We're missing the code to set `loading=true` when retrieving the list of devices

**Review by knolleary — CHANGES_REQUESTED** [review]

**Comment by joepavitt** [comment]
@knolleary fixed the loading variable setting in devices accordingly.

**Pull request merged**